Automattic/jetpack
# Premium blocks: Show notification on plan upgrade/checkout

Fixes #13003 (partially)

#### Changes proposed in this Pull Request:


Adds a ~`plan`~ `plan_upgraded` query parameter handler for the editor pages that will dispatch a notification when observed. Effectively this will be passed on redirection from WPCOM once a plan has been upgraded successfully after clicking on the "upgrade nudge".

The approach and pretty much everything but the action dispatch have been discussed & agreed among @simison & @ockham in Slack: p1565251913019200-slack-luna

The result is a regular "success" notice shown in the editor if the parameter is set e.g. `http://foobar.ngrok.io/wp-admin/post.php?post=1&action=edit&plan_upgraded=123`

<img width="1280" alt="Screenshot 2019-08-10 at 2 24 51 PM" src="https://user-images.githubusercontent.com/1705499/62821373-4d603400-bb7c-11e9-927f-5ec31cf431ff.png">

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Yes & no. It adds the foundation for showing a notification when the redirection happens after clicking an upgrade nudge.

#### Testing instructions:






* Checkout the underlying branch.
* Add the parameter `&plan_upgraded=true` to an editor page e.g. `http://foobar.ngrok.io/wp-admin/post.php?post=1&action=edit&plan_upgraded=true` (or localhost).
* Observe that the result is a dismissible notice, as in the screenshot above.

#### Proposed changelog entry for your changes:

* Doesn't need one I think?
